### PR TITLE
Fixed regression in wait on jobs

### LINKED ISSUE #3025: [regression] `rm -i` prompt is failing (again)
### xonfig

<details>

```
$ xonfig
+------------------+-----------------+
| xonsh            | 0.8.10          |
| Python           | 3.7.2           |
| PLY              | 3.11            |
| have readline    | True            |
| prompt toolkit   | 2.0.8           |
| shell type       | prompt_toolkit2 |
| pygments         | 2.3.1           |
| on posix         | True            |
| on linux         | True            |
| distro           | arch            |
| on darwin        | False           |
| on windows       | False           |
| on cygwin        | False           |
| on msys2         | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
+------------------+-----------------+
```

</details>

## issue

Looked through the old issues and it seems like this problem should be solved. So this might be a regression.

```
$ xonsh --no-rc
$ touch file1 file2 file3 file4
$ rm -i file*
/usr/bin/rm: remove regular empty file 'file1'? /usr/bin/rm: remove regular empty file 'file2'? /usr/bin/rm: remove regular empty file 'file3'? /usr/bin/rm: remove regular empty file 'file4'? /usr/bin/rm: error closing file
```

I checked that `rm` is not aliased to anything else in Bash. I also tried with `xonsh --no-rc --shell-type readline` with the same result. Finally, setting `$XONSH_STORE_STDOUT` to True or False makes no difference.

### versions

<details>

```
$ rm --version
rm (GNU coreutils) 8.30
$ uname -a
Linux 4.20.8-arch1-1-ARCH #1 SMP PREEMPT Wed Feb 13 00:06:58 UTC 2019 x86_64 GNU/Linux
```

</details>

### PR DESCRIPTION
**Pull request by scopatz**
Fixes #3025

### CODE REVIEW

**Comment by melund** [comment]
Not sure I understand the logic here. @mogenson does it fix your issue?

File: xonsh/jobs.py [hunk 1]
```diff
@@ -219,6 +219,20 @@ def wait_for_active_job(last_task=None, backgrounded=False, _nochild=False):
...
+    """Safely call wait_for_active_job()"""
+    have_error = True
+    while have_error:
+        try:
+            rtn = wait_for_active_job(
+                last_task=last_task, backgrounded=backgrounded, return_error=True
+            )
+        except ChildProcessError as e:
+            rtn = e
+        have_error = isinstance(e, ChildProcessError)
```
Comment (melund) [inline xonsh/jobs.py]: There is an issue here; `e` may not exist if an exception wasn't thrown. Should this have been indented? 

But anyway I don't think I understand things here. What is the reasoning behind `_safe_wait_for_active_job` and `wait_for_active_job` calling each other?

Comment (melund) [reply xonsh/jobs.py]: Or should it be `have_error = isinstance(rtn, ChildProcessError)`

Comment (scopatz) [reply xonsh/jobs.py]: Oh yeah, it should have been rtn. Good catch!

**Comment by scopatz** [comment]
@melund - I think this is ready for another review!

**Comment by scopatz** [comment]
pinging @xore :)

**Comment by melund** [comment]
Sorry. I have been a bit preoccupied to day. This looks good.

**Comment by scopatz** [comment]
No worries thanks @melund! I just want to cut a release soon :wink:

**Pull request merged**